File: Divide_Array_Into_Arrays_With_Max_Difference.java
class Solution {
    public int[][] divideArray(int[] nums, int k) {
        Arrays.sort(nums);
        int[][] res = new int[nums.length / 3][3];
        
        for (int i = 0; i < nums.length; i += 3) {
            if (nums[i] + k < nums[i + 2]) {
                return new int[0][0];
            }

            res[i/3] = new int[]{nums[i], nums[i + 1], nums[i + 2]};
        }
        
        return res;     
    }
}

File: Divide_Array_Into_Arrays_With_Max_Difference.cpp
class Solution {
public:
    vector<vector<int>> divideArray(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        vector<vector<int>> res;

        for (int i = 0; i < nums.size(); i += 3) {
            if (nums[i] + k < nums[i + 2]) {
                return {};
            }

            res.push_back({nums[i], nums[i + 1], nums[i + 2]});
        }
        
        return res;      
    }
};
